atef/widgets/utils.py

```diff
@@ -216,3 +216,93 @@ def reset_cursor(self):
         if self.ignore_events:
             self.app = QtWidgets.QApplication.instance()
             self.app.removeEventFilter(FILTER)
+
+
+def _create_vbox_layout(
+    widget: Optional[QtWidgets.QWidget] = None, alignment: Qt.Alignment = Qt.AlignTop
+) -> QtWidgets.QVBoxLayout:
+    if widget is not None:
+        layout = QtWidgets.QVBoxLayout(widget)
+    else:
+        layout = QtWidgets.QVBoxLayout()
+    layout.setAlignment(alignment)
+    return layout
+
+
+class ExpandableFrame(QtWidgets.QFrame):
+    """
+    A `QtWidgets.QFrame` that can be toggled with a mouse click.
+
+    Contains a QVBoxLayout layout which can have one or more user-provided
+    widgets.
+
+    Parameters
+    ----------
+    text : str
+        The title of the frame, shown on the toolbutton.
+
+    parent : QtWidgets.QWidget, optional
+        The parent widget.
+    """
+
+    toggle_button: QtWidgets.QToolButton
+    _button_text: str
+    _size_hint: QtCore.QSize
+
+    def __init__(self, text: str = "", parent: Optional[QtWidgets.QWidget] = None):
+        super().__init__(parent=parent)
+
+        self._button_text = text
+
+        self.toggle_button = QtWidgets.QToolButton(text=text)
+        self.toggle_button.setCheckable(True)
+        self.toggle_button.setChecked(False)
+        self.toggle_button.setStyleSheet("QToolButton { border: none; }")
```

Comment:
Aside: we should probably move to making an atef stylesheet at some point and move these things there (`atef.qss`, oh yeah!)

Reply:
I wonder if that'd fix some of the weird osx rendering differences 🤔 A good idea, but I'm not volunteering right now... 👀